### DISCUSSION TITLE
Masking an image in Scratch

### DISCUSSION THREAD

**Discussion by damightyq** [ScratchAddons/ScratchAddons#5350] [Ideas]
Can you guys make a tool that allows masking something a sprite or a costume in Scratch? It would be really usefull for health bars and hiding unwanted stuff as well. For example: 

Masking Off:
![resultsgaugeawhellnaw](https://user-images.githubusercontent.com/102260104/201559207-6f659808-8fb2-49da-82f6-96e6fcedb44d.png)

Masking On:
![resultsgaugefull](https://user-images.githubusercontent.com/102260104/201559309-bd79c9de-373e-45f2-9b75-54580c0914cd.png)

It is really usefull this way so that I don't have to put in in frame by frame when it's going up.

**Comment by BroJac5246** [comment]
I'm not sure if I'm understanding correctly, but could you just use the fill bucket outside the object and set the color to transparent?

**Reply by Squatterrr** [reply]
Masking could do so much more since it is done within code. Would you rather create thousands of costumes for a cool animation, or would you rather just use masking to run it easily

**Comment by DNin01** [comment]
If we were to provide this feature, we need to make sure that those who do not have Scratch Addons installed can still run the same projects. If this addon would add more blocks to Scratch, then users running the Scratch projects with this feature will have to be using SA for it to work as intended.

If this is a costume editor enhancement, though, then it should be fine.

**Reply by damightyq** [reply]
It would be useful in the costume editor, but with loading bars that are in a weird shape, I could just recommend to get SA in the instructions.

**Comment by Endermaster3690** [comment]
scratch 2.0 had a wand tool like this didn't it

**Reply by damightyq** [reply]
I've never used 2.0 sorry... But I can imagine it was really useful in 2.0

**Reply by damightyq** [reply]
Mabye they can just rewrite the code to make it work for 3.0

**Reply by Secret-chest** [reply]
> Mabye they can just rewrite the code to make it work for 3.0

Scratch 2 used Flash, Scratch 3 uses JavaScript

**Reply by damightyq** [reply]
I'm not a JavaScript expert (if you can't tell I'm a scratcher lol) but is it possible to code masking into JavaScript?

**Reply by BroJac5246** [reply]
> I'm not a JavaScript expert (if you can't tell I'm a scratcher lol) but is it possible to code masking into JavaScript?

You can do almost anything with JavaScript, but it's often the complexity to code a feature that holds back those good ideas that never get added.
